src/dataset-builder/DatasetBuilder.test.ts

```diff
@@ -391,30 +391,8 @@ describe('DatasetBuilder', () => {
     initializeValidDatasetRequest();
     await user.click(await screen.findByText('Request this data snapshot'));
     // Assert
-    expect(await screen.findByText('Access request created in Terra')).toBeTruthy();
-  });
-
-  it('Calls createSnapshotAccessRequest with the correct body', async () => {
-    const createSnapshotAccessRequest = jest.fn().mockResolvedValue(mockCreateSnapshotAccessRequest);
-    const mockDataRepoContract: Partial<DataRepoContract> = {
-      snapshot: (_snapshotId) =>
-        ({
-          getSnapshotBuilderCount: () => Promise.resolve({ result: { total: 19 }, sql: '' }),
-        } as Partial<DataRepoContract['snapshot']>),
-      snapshotAccessRequest: () =>
-        ({
-          createSnapshotAccessRequest,
```

Comment:
This was the original bug; this test was mocking `snapshotAccessRequest` but only provided `createSnapshotAccessRequest`. Since it didn't provide a mock for `getSnapshotAccessRequestDetails`, the render would eventually fail. Once I added the same mock as on line 383, the test setup code was identical so I merged the test cases.